### DISCUSSION TITLE
Solved - First deploy HTTP 500 with Lumen

### DISCUSSION THREAD

**Discussion by richoandika** [brefphp/bref#1143] [Support] [answered]
I have problem using bref on Lumen v8.0, but in local that code working as normal.
Im using PHP 7.4.21 in local and serverless.yml same too
These the errors from serverless logs:
<pre>
START RequestId: 2cc8d7a8-5111-4181-9173-8e1c9f86a481 Version: $LATEST
[23-Jan-2022 12:35:47] NOTICE: fpm is running, pid 9
[23-Jan-2022 12:35:47] NOTICE: ready to handle connections
NOTICE: PHP message: PHP Fatal error:  Uncaught UnexpectedValueException: There is no existing directory at "/var/task/storage/logs" and it could not be created: Read-only file system in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:216
Stack trace:
#0 /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php(135): Monolog\Handler\StreamHandler->createDir()
#1 /var/task/vendor/monolog/monolog/src/Monolog/Handler/RotatingFileHandler.php(125): Monolog\Handler\StreamHandler->write()
#2 /var/task/vendor/monolog/monolog/src/Monolog/Handler/AbstractProcessingHandler.php(48): Monolog\Handler\RotatingFileHandler->write()
#3 /var/task/vendor/monolog/monolog/src/Monolog/Logger.php(327): Monolog\Handler\AbstractProcessingHandler->handle()
#4 /var/task/vendor/monolog/monolog/src/Monolog/Logger.php(565): Monolog\Logger->addRecord()
#5 /var/task/vendor/illuminate/log/Logger.php(183): Monolog\Logger->error()
#6 /var/task/vendor/illuminate/log/Logger.php(94): Illuminate\Log\Logger->writeLog()
#7 /var/task/vendor/illuminate/lo in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php on line 216
NOTICE: PHP message: PHP Fatal error:  Uncaught UnexpectedValueException: There is no existing directory at "/var/task/storage/logs" and it could not be created: Read-only file system in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:216
Stack trace:
#0 /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php(135): Monolog\Handler\StreamHandler->createDir()
#1 /var/task/vendor/monolog/monolog/src/Monolog/Handler/RotatingFileHandler.php(125): Monolog\Handler\StreamHandler->write()
#2 /var/task/vendor/monolog/monolog/src/Monolog/Handler/AbstractProcessingHandler.php(48): Monolog\Handler\RotatingFileHandler->write()
#3 /var/task/vendor/monolog/monolog/src/Monolog/Logger.php(327): Monolog\Handler\AbstractProcessingHandler->handle()
#4 /var/task/vendor/monolog/monolog/src/Monolog/Logger.php(565): Monolog\Logger->addRecord()
#5 /var/task/vendor/illuminate/log/Logger.php(183): Monolog\Logger->error()
#6 /var/task/vendor/illuminate/log/Logger.php(94): Illuminate\Log\Logger->writeLog()
#7 /var/task/vendor/illuminate/lo in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php on line 216
END RequestId: 2cc8d7a8-5111-4181-9173-8e1c9f86a481
REPORT RequestId: 2cc8d7a8-5111-4181-9173-8e1c9f86a481  Duration: 267.50 ms     Billed Duration: 607 ms Memory Size: 1024 MB    Max Memory Used: 79 MB        Init Duration: 338.92 ms
nd it could not be created: Read-only file system in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:216
Stack trace:
#0 /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php(135): Monolog\Handler\StreamHandler->createDir()
#1 /var/task/vendor/monolog/monolog/src/Monolog/Handler/RotatingFileHandler.php(125): Monolog\Handler\StreamHandler->write()
#2 /var/task/vendor/monolog/monolog/src/Monolog/Handler/AbstractProcessingHandler.php(48): Monolog\Handler\RotatingFileHandler->write()
#3 /var/task/vendor/monolog/monolog/src/Monolog/Logger.php(327): Monolog\Handler\AbstractProcessingHandler->handle()
#5 /var/task/vendor/illuminate/log/Logger.php(183): Monolog\Logger->error()
#6 /var/task/vendor/illuminate/log/Logger.php(94): Illuminate\Log\Logger->writeLog()
#7 /var/task/vendor/illuminate/lo in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php on line 216
Stack trace:
#0 /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php(135): Monolog\Handler\StreamHandler->createDir()
#1 /var/task/vendor/monolog/monolog/src/Monolog/Handler/RotatingFileHandler.php(125): Monolog\Handler\StreamHandler->write()
#2 /var/task/vendor/monolog/monolog/src/Monolog/Handler/AbstractProcessingHandler.php(48): Monolog\Handler\RotatingFileHandler->write()
#4 /var/task/vendor/monolog/monolog/src/Monolog/Logger.php(565): Monolog\Logger->addRecord()
#5 /var/task/vendor/illuminate/log/Logger.php(183): Monolog\Logger->error()
#6 /var/task/vendor/illuminate/log/Logger.php(94): Illuminate\Log\Logger->writeLog()
#7 /var/task/vendor/illuminate/lo in /var/task/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php on line 216
END RequestId: 7c46413a-8750-4b69-8880-d1aac2349da5
REPORT RequestId: 7c46413a-8750-4b69-8880-d1aac2349da5  Duration: 5.96 ms       Billed Duration: 6 ms   Memory Size: 1024 MB    Max Memory Used: 79 MB
</pre>

**Comment by t-richard** [comment]
Hi :wave: 

The error says 

> There is no existing directory at "/var/task/storage/logs" and it could not be created: Read-only file system

On Lambda, the file system is readonly and logs should be forwarded to Cloudwatch (the AWS service responsible for logging & monitoring). Bref automatically forwards everything written to `php://stderr` to Cloudwatch so you need to configure Monolog to write logs there.

I'm not familiar with Lumen so I can't give any precise guidance here but for reference, [here's what Bref does in the Laravel Bridge ](https://github.com/brefphp/laravel-bridge/blob/36111f777e78759c2f20e9555dea2cd92e606916/src/BrefServiceProvider.php#L58-L61)

**Reply by richoandika** [reply]
Hi, @t-richard 👋
Thanks for response, but anyway I don't really understand about that.
Trying to find solution but I found the bug is in register middleware and service providers. At the moment I comment the any package, it's work! I don't how the fix that..
Can somebody help me??

**Comment by richoandika** [comment, marked as answer]
btw i solved the problem
first, make sure the package. you can remove vendor and reinstall again.
then flush the application cache ..
